### PR TITLE
Refactoring hyku to move fedora and solr into separate containers

### PR DESCRIPTION
**Pull request by astrilet**
Refactoring hyku to separate solr and fedora into separate containers

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,83 +1,70 @@
...
+    volumes:
+      - fedora:/fedora
+#    ports:
+#      - 8080:8080
+
+# This image actually runs hyku app
+  web:
     image: ualibraries/hyku_app
+    command: /usr/local/bin/start_hyku.sh
+    env_file: .env
```
Comment (murny) [inline docker-compose.yml]: Why have some env vars come from `.env` and others being set in `environment:`? Choose one and stick with it?

File: docker-compose.yml [hunk 2]
```diff
@@ -1,83 +1,70 @@
...
-      - 6379:6379
+#    ports:
+#      - 6379:6379
+#
+# SOLR indexing
+  solr:
+    image: ualibraries/solr_hydra
+    command: /usr/local/bin/start_solr.sh
+    environment:
+      - SOLR_CORE=production
```
Comment (murny) [inline docker-compose.yml]: What is this environment variable doing and why is it needed? Don't see anything that needs this in the default solr docker image?

File: docker-compose.yml [hunk 3]
```diff
@@ -1,83 +1,70 @@
...
-    networks:
-      internal:
-    ports:
-      - 6379:6379
+#    ports:
+#      - 6379:6379
+#
+# SOLR indexing
+  solr:
+    image: ualibraries/solr_hydra
```
Comment (murny) [inline docker-compose.yml]: Where does this live? I assume it isn't this one: https://github.com/ualbertalib/di_docker_images/tree/master/solr?

As this is putting HN2 config in the solr? which I assume is for https://hub.docker.com/r/ualibraries/hn2_solr/

File: README.md [hunk 4]
```diff
@@ -1,31 +1,107 @@
-This is docker image to try out Hyku implementation for evaluation purposes only.
+# This is project to get [Hyku](https://github.com/projecthydra-labs/hyku.git) running in docker environment for evaluation purposes only.
 
 
-There are still a few bugs to work out at this time, but image is stable an workable (I hope... )
+There are two docker-compose files dockre-compose.yml and docker-compose-setup.yml.
+You will need to run docker-compose-setup.yml once before very fist time you start Hyku to setup database and
```
Comment (murny) [inline README.md]: Why? Why do you need a setup docker container that just setups the database??? This is same thing as the HN2 docker compose review: https://github.com/ualbertalib/didocker_hydranorth2/pull/2#discussion_r109213840

File: docker-compose-setup.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+version: '2.1'
```
Comment (murny) [inline docker-compose-setup.yml]: This entire file shouldn't be needed...see above: https://github.com/ualbertalib/didocker_hyku/pull/11/files#r114024453

File: files/start_hyku.sh [hunk 6]
```diff
@@ -3,12 +3,6 @@
 echo "Cleaning tmp directory"
 rm -rf /data/tmp/*
 
-echo "Starting Solr..."
-/usr/local/bin/start_solr.sh &
-
-echo "Starting Fedora..."
-/usr/local/bin/start_fedora.sh &
-
 echo "Starting sidekiq..."
```
Comment (murny) [inline files/start_hyku.sh]: Sidekiq should be a different docker container

File: files/start_hyku.sh [hunk 7]
```diff
@@ -3,12 +3,6 @@
 echo "Cleaning tmp directory"
```
Comment (murny) [inline files/start_hyku.sh]: Still dont see the need for these scripts....run them as docker commands or have this setup live in the dockerfile.

Theres also a `sleep 15` here. Why? 15 seconds to sleep in a startup script is crazy?

File: docker-compose.yml (hunk 3 above)
Comment (murny) [inline docker-compose.yml]: This needs to be addressed...still a pending issue in HN2 docker that wasn't tackled.

https://github.com/ualbertalib/didocker_hydranorth2/pull/2#discussion_r109217455

File: files/start_hyku.sh (hunk 7 above)
Comment (murny) [inline files/start_hyku.sh]: https://github.com/ualbertalib/didocker_hydranorth2/pull/2#discussion_r109217793

File: docker-compose.yml [hunk 8]
```diff
@@ -1,83 +1,70 @@
...
+#      - 8983:8983
 
-  app:
+# Fedora object store
+  fedora:
+    image: ualibraries/fedora
+    volumes:
+      - fedora:/fedora
+#    ports:
+#      - 8080:8080
```
Comment (murny) [inline docker-compose.yml]: https://github.com/ualbertalib/didocker_hydranorth2/pull/2#discussion_r109218740

File: docker-compose-setup.yml [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+      timeout: 5s
+      retries: 100
+
+    volumes:
+      - db:/var/lib/postgresql/data
+
+  setup:
+    image: ualibraries/hyku_app
+    env_file: .env
+    command: /bin/bash -c "if [ ! -e /db/initialized ]; then sleep 25 && /data/bin/setup && touch /db/initialized; fi"
```
Comment (murny) [inline docker-compose-setup.yml]: See comments here....assuming this is for setting up the DB....you should be able to run db:setup/migrate as many times as you want?
https://github.com/ualbertalib/didocker_hydranorth2/pull/2#discussion_r109983477
https://github.com/ualbertalib/didocker_hydranorth2/pull/2#discussion_r109983568

File: docker-compose.yml (hunk 1 above)
Comment (astrilet) [inline docker-compose.yml]: common variable are in .env file container specific is in environment: section

File: docker-compose.yml (hunk 2 above)
Comment (astrilet) [inline docker-compose.yml]: it is used by solr_hydra image to create core.  We might want to have multiple solr cores for development and testing.

Comment (murny) [inline docker-compose.yml]: Still not seeing this being used in the https://github.com/ualbertalib/di_docker_images/tree/master/solr repo? Would it be possible to give me a github link of this code?

File: docker-compose.yml (hunk 1 above)
Comment (murny) [inline docker-compose.yml]: What makes something a common variable versus a container specific variable? I see rails config in both .env and environment section. I also see database/infrastructure config in both sections as well? Can I get more details here between the difference or why the difference needs to exist?

**Comment by astrilet** [comment]
We do not have dev environment on hyku as we do not own/develop it.

**Comment by weiweishi** [comment]
@astrilet close the PR per our discussion. Please feel free to add your comment to summarize what was included in this PR that you think could be useful for us in the future.

**Pull request closed without merging**